casper/src/main/resources/MultiSigRevVault.rho

```diff
@@ -0,0 +1,215 @@
+ 4.  | 1,         | secp256k1          | pk = 04fe2eb1e0e7462b1a8f64600389e1e76727f8b2d38804eaa4b48f7a7d6715130fc24d3c4dac2d8bdc19e0b49879dbaf07c30773cd9740a9d14a092ef76339207a
+ 5.  | 4, 2,      | genIds             | uname = Unforgeable(0x022e67c5237b24bcc1971f00556aa180f8d864c67e23989c9d550f8c9265cb53)
+ 6.  | 3, 5,      | registry           | value = (9223372036854775807, bundle+ {   Unforgeable(0x022e67c5237b24bcc1971f00556aa180f8d864c67e23989c9d550f8c9265cb53) })
+ 7.  | 6,         | protobuf           | toSign = 2a40aa013d0a0d2a0b10feffffffffffffffff010a2c5a2a0a263a240a220a20022e67c5237b24bcc1971f00556aa180f8d864c67e23989c9d550f8c9265cb531001
+ 8.  | 7, 1,      | secp256k1          | sig = 30450221008539caafe08b2a27cfcabdcf9c7c345a1c1a926330a1a320ca614cf9240f6c77022024ad8465597c8a673431007b8e33317bd49c9c68dacb41eef6a4316d933909c4
+ 9.  | 4,         | registry           | uri = rho:id:b9s6j3xeobgset4ndn64hje64grfcj7a43eekb3fh43yso5ujiecfn
+ ----+------------+--------------------+-----------------------------------------------------------------------------------------------------------------------------------------------------
+ */
+
+new MultiSigRevVault,
+    rs(`rho:registry:insertSigned:secp256k1`), uriOut,
+    rl(`rho:registry:lookup`), 
+    _fromPublicKey,
+    _multiSigRevVault,
+    RevVaultCh,
+    AuthKeyCh,
+    ListOpsCh
+in {
+  rl!(`rho:lang:listOps`, *ListOpsCh) |
+  rl!(`rho:rchain:authKey`, *AuthKeyCh) |
+  rl!(`rho:rchain:revVault`, *RevVaultCh) |
+  for(@(_, ListOps) <- ListOpsCh;
+      @(_, AuthKey) <- AuthKeyCh;
+      @(_, RevVault) <- RevVaultCh) {
+
+    contract MultiSigRevVault(@"foo", ret) = { ret!("foo") } |
+
+    contract MultiSigRevVault(@"deployerAuthKey", deployerId, ret) = {
+      new RevAddress(`rho:rev:address`), 
+          DeployerIdOps(`rho:rchain:deployerId:ops`), 
+          revAddrCh, 
+          deployerPubKeyBytesCh in {
+        DeployerIdOps!("pubKeyBytes", *deployerId, *deployerPubKeyBytesCh) |
+        for (@deployerPubKeyBytes <- deployerPubKeyBytesCh) {
+          _fromPublicKey!(deployerPubKeyBytes, *ret)
+        }
+      }
+    } |
+    
+    contract MultiSigRevVault(@"unforgeableAuthKey", unf, ret) = {
+      new RevAddress(`rho:rev:address`), revAddrCh in {
+        RevAddress!("fromUnforgeable", *unf, *revAddrCh) |
+        for (@unfRevAddress <- revAddrCh) {
+          @AuthKey!("make", (*_multiSigRevVault, unfRevAddress), *ret)
+        }
+      }
+    } |
+
+    contract _fromPublicKey(@publicKey, ret) = {
+      new RevAddress(`rho:rev:address`), revAddrCh in {
+        RevAddress!("fromPublicKey", publicKey, *revAddrCh)|
+        for (@revAddress <- revAddrCh) {
+          @AuthKey!("make", (*_multiSigRevVault, revAddress), *ret)
+        }
+      }
+    } |
+    
+    // MultiSig
+    // Returns (false, errorMsg) or (true, (multiSigVault, revAddress, revVault))
+    contract MultiSigRevVault(@"create", @publicKeys, @authKeys, @quorumSize, ret) = {
```

Comment:
Having two parameters (`publicKeys` and `authKeys`) where one is always unused might be confusing to users.
I agree though that the alternatives are suboptimal as well (using an Either or creating two methods: `createFromPks` and `createFromAuthKeys`).
I would probably slightly prefer the alternative where we create two methods.

Reply:
Either or both can be used.